arrow-fx/src/main/kotlin/arrow/fx/Queue.kt

```diff
@@ -20,95 +19,244 @@ typealias QueuePartialOf<F> = Kind<ForQueue, F>
 inline fun <F, A> QueueOf<F, A>.fix(): Queue<F, A> =
   this as Queue<F, A>
 
+/** A polymoprhic effect typeclass that allows [Dequeue]'ing values from a [Queue]. */
 interface Dequeue<F, A> {
+
+  /**
+   * Take a value from the [Queue], or sementically blocks until a value becomes available.
+   *
+   * @see [peek] for a function that doesn't remove the value from the [Queue].
+   * @see [tryTake] for a function that does not sementically block but returns immediately with an [Option].
+   */
   fun take(): Kind<F, A>
+
+  /**
+   * Tries to take a value from the [Queue]. Returns immediately with either [None] or a value [Just].
+   *
+   * @see [take] for function that sementically blocks until a value becomes available.
+   * @see [tryPeek] for a function that sementically blocks until a value becomes available.
+   */
+  fun tryTake(): Kind<F, Option<A>>
+
+  /**
+   * Peeks a value from the [Queue] or semantically blocks until one becomes available.
+   * In contrast to [take], [peek] does not remove the value from the [Queue].
+   *
+   * @see [tryPeek] for a function that does not sementically blocks but returns immediately with an [Option].
+   */
+  fun peek(): Kind<F, A>
+
+  /**
+   * Tries to peek a value from the [Queue]. Returns immediately with either [None] or a value [Just].
+   * In contrast to [tryTake], [tryPeek] does not remove the value from the [Queue].
+   *
+   * @see [peek] for a function that sementically blocks until a value becomes available.
+   */
+  fun tryPeek(): Kind<F, Option<A>>
+
+  /**
+   * Takes all values currently available in the [Queue] and empties the [Queue] or returns an empty list.
+   *
+   * It also takes all outstanding offers, in case of a blocking strategy such as bounded.
+   */
+  fun takeAll(): Kind<F, List<A>>
+
+  /**
+   * Peeks all values currently available in the [Queue] or returns an empty list.
+   */
+  fun peekAll(): Kind<F, List<A>>
 }
 
+/** A polymoprhic effect typeclass that allows [Dequeue]'ing values from a [Queue]. */
 interface Enqueue<F, A> {
+
+  /**
+   * Offers a value to the [Queue], and might behave differently depending on the [Queue.BackpressureStrategy].
+   *
+   *  - Sementically blocks until room available in [Queue] for [Queue.BackpressureStrategy.Bounded]
+   *  - Returns immediately and slides values through the [Queue] for [Queue.BackpressureStrategy.Sliding]
```

Comment:
Not sure I understand the meaning of "slide" when talking about offering, maybe an explanation like https://github.com/arrow-kt/arrow-fx/pull/30/files#diff-f560b92378f730cad2c0c8916a5e0071R105 would be more descriptive?

Reply:
I'm going to improve on this before merging, I'll poke you for another tech content review.